### PR TITLE
Fix `C3DTilesLayer` and `C3DTFeature`

### PR DESCRIPTION
**Pull request by valentinMachado**
## Description
`C3DTFeature` reference the `THREE.Object3D` where they belong.
`computeWorldBox3` is now a method of `C3DTFeature` possible due to the latter modification
Some code clean.
`updateStyle` is call after dispacthing event `ON_TILE_CONTENT_LOADED`

## Motivation and Context
`C3DTFeature` reference the `THREE.Object3D` where they belong.
=> tileContent can have different child with feature in it which was not handle by the previous code
`updateStyle` is call after dispacthing event `ON_TILE_CONTENT_LOADED`
=> when listener of this event influences `C3DTilesLayer` style the user does not have to call an extra `updateStyle` of the layer

waiting this to be approved to rename commits name

### CODE REVIEW

**Review by jailln — APPROVED** [review]
Ok for me 👍 thanks !

**Comment by valentinMachado** [comment]
Great ! I've made a single commit, and CI is working.

File: src/Core/3DTiles/C3DTFeature.js [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
  * @property {Array<{start:number,count:number}>} groups - groups in geometry.attributes matching batchId
  * @property {object} info - info in the batchTable
  * @property {object} [userData={}] - some userData
  */
 class C3DTFeature {
     #info;
-    constructor(tileId, batchId, groups, info, userData = {}) {
-        /** @type {number} */
-        this.tileId = tileId;
+    constructor(tileId, object3d, batchId, groups, info, userData = {}) {
```
Comment (jailln) [inline src/Core/3DTiles/C3DTFeature.js]: I just realized that this class is exposed in the API. Can you handle the case when users use the old API, log a deprecation message and add a BREAKING CHANGE message in the body of your commit please ?

Comment (valentinMachado) [reply src/Core/3DTiles/C3DTFeature.js]: Done

**Review by jailln — APPROVED** [review]

**Pull request merged**